### PR TITLE
Day 4

### PR DESCRIPTION
**Pull request by josejorgers**
* Add day 4
* Modified structure of the repo for better organization and correct test running

The problem with the previous approach is that when trying to test the project, Pytest also collected the test for day 4. Then it is better to separate the tutorial from the project itself

### CODE REVIEW

File: tutorial/guias/Paso_a_paso_DIA_4.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+class TestFruit(TestCase):
+    def setUp(self):
+        self.fruit = Fruit((10, 10))
+
+    def test_created_as_dataclass(self):
+        self.assertEqual(self.fruit.position, (10, 10))
+
+```
+
+Analicemos el código anterior línea a línea. En la primera línea estamos importantdo la clase `TestCase` del módulo `unittest` que viene incluido en Python. Esta clase contiene lo necesario para crear pruebas automatizadas. Lo que haremos será heredar de ella y crear nuestras propias pruebas.
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_4.md]: Typo: `importantdo` --> `importando`

File: tutorial/guias/Paso_a_paso_DIA_4.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+class TestFruit(TestCase):
+    def setUp(self):
+        self.fruit = Fruit((10, 10))
+
+    def test_created_as_dataclass(self):
+        self.assertEqual(self.fruit.position, (10, 10))
+    
+    def test_fail(self):
+        self.assertEqual(1, 2)
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_4.md]: Maybe add a comment in the code here to highlight that this is a dummy function just for showing how tests fail

File: tutorial/guias/Paso_a_paso_DIA_4.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+3. La función `test_trim` prueba que el tamaño de la serpiente disminuya de manera correcta. Luego de ejecutarse, el cuerpo de la serpiente debe ser `[(50, 50), (40, 50), (30, 50)]`. Debemos tener en cuenta que para cada test la serpiente que se utiliza como entrada es la que se crea en el método `setUp` y no la que resulta de ejecutar el test anterior.
+4. La función `test_change_direction` verifica que la dirección se cambie de manera correcta. En este caso se intenta cambiar la dirección hacia arriba y se espera que, en efecto, la nueva dirección de la serpiente pase a ser hacia arriba :arrow_up:.
+5. Por último la función `test_change_direction_opposite` verifica que la dirección de la serpiente no cambie cuando se intenta moverla en la dirección opuesta. Como inicialmente la serpiente se mueve hacia la derecha, cambiar la posición hacia la izquierda no debiera tener ningún efecto y la serpiente debiera mantener su rumbo hacia la derecha :arrow_right:.
+
+>:warning: Ten siempre en cuenta que el método `setUp` se ejecuta antes de cada test por lo que cada test usa la misma serpiente como entrada.
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_4.md:182]: I don't know if we should add a small Note stating the importance that tests should be written without taking the result from other tests into consideration, as a good practice.

File: tutorial/guias/Paso_a_paso_DIA_4.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+from unittest import TestCase
+
+from src.model.snake import Direction, Snake
+
+
+class TestSnake(TestCase):
+    def setUp(self):
+        self.snake = Snake(position=(50, 50), direction=Direction.RIGHT, size=4)
+
+    def test_create_body(self):
```
Comment (albexl) [inline tutorial/guias/Paso_a_paso_DIA_4.md:148]: Should we add a note saying that usually these tests should start with the prefix `test`. I think that can be modified to either start with `test` or `test_`, or even end with `test` or `_test`. But the default I think it is starting with `test`.

**Review by albexl — CHANGES_REQUESTED** [review]
Looks good to me. Read my comments and fix the small typo. It is up to you if we should add the things I mentioned.

File: tutorial/guias/Paso_a_paso_DIA_4.md (hunk 4 above)
Comment (josejorgers) [reply tutorial/guias/Paso_a_paso_DIA_4.md:148]: I say something similar above when talking about snake_casing. I don't know if adding a note would be too repetitive.

**Pull request merged**